PyRoute/TradeCodes.py

```diff
@@ -26,6 +26,53 @@ class TradeCodes(object):
     pcolor = {'As': '#8E9397', 'De': '#d17533', 'Fl': '#e37dff', 'He': '#ff6f0c', 'Ic': '#A5F2F3',
               'Oc': '#0094ED', 'Po': '#6a986a', 'Va': '#c9c9c9', 'Wa': '#4abef4'}
     ext_codes = {'Lt', 'Ht', 'Lg', 'Hg'}
+    weird_codes = {'{Anomaly}', '{Fuel}', '{Ringworld}', '{Rosette}'}
+    allowed_residual_codes = {'Ag', 'Ba', 'Bo', 'C0', 'C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7', 'C8', 'C9', 'Cl', 'Cw',
+                              'Cy', 'D0', 'D1', 'D2', 'D3', 'D4', 'D5', 'D6', 'D7', 'D8', 'D9', 'Dw', 'Ex', 'Fr', 'Ga',
+                              'Hi', 'In', 'Lo', 'N1', 'Na', 'Ni', 'o', 'Pa', 'Ph', 'Pi', 'Po', 'Pr', 'Ri', 'Rn', 'Rv',
+                              's', 'Sp', 'St', 'Tn', 'Za'}
+    ok_pairs = {('Ni', 'Po'), ('Ni', 'Pa'), ('Ni', 'Wa'), ('Ph', 'Ri'), ('Lo', 'Va'), ('Ni', 'Va'), ('Lo', 'Po'),
```

Comment:
This seems fragile to me. It's also difficult to review. 
1. Put this into alphabetical order. Keep in mind the trade codes (in the sector files) should always be in alphabetical order. So you can have (As, Va) but never (Va, As). 
2. The only codes that need to be in the list are the `pcode` (planetary codes) list above. You don't need to include the population codes (Ba, Lo, Ni, Ph, Hi) as they can go with any of the Pcodes.

Reply:
Also the `Cx` and `Dx` (where x in `0-9w`) could also be removed.

Reply:
Again, this was summarising existing practice in the TravMap data, for much the same reason as adding ParseStarInput.

Reply:
`TradeCode.ok_pairs` is now alphabetised.